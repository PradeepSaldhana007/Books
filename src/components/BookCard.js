import React, { Component } from 'react';
import { Card, Button } from 'react-bootstrap';

class BookCard extends Component {
    render() {
        return (
            <Card style={{ width: '18rem', height: '12rem' }}>
                <Card.Body>
                    <Card.Title>{this.props.title}</Card.Title>
                    <Card.Text>
                        Author: {this.props.author} Price: {this.props.price}
                    </Card.Text>

                    {/* <Button variant="primary" id="addButton" onClick={this.addCart}>Add To Cart</Button> */}

                    {
                       this.props.action === 'add' ?  
                       <Button variant="primary" id="addButton" onClick={this.props.addToCartClick}>Add To Cart</Button> : 
                       <Button variant="danger" id="deleteButton" onClick={this.props.deleteFromCartClick}>Delete</Button>
                    }
                    
                </Card.Body>
            </Card>
        );
    }

    // addCart = () => {
    //     this.props.addToCartClick();
    //     console.log('added');
    // }
}

export default BookCard;