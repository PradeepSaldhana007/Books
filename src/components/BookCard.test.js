import React from "react";
import { shallow } from "enzyme";
import BookCard from "./BookCard";
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe("Executes a handler function", () => {
  it("must call the mock method with button click", () => {
    const actionAdd = "add";
    const wrapper = shallow(<BookCard action={actionAdd} />);
    const button = wrapper.find("#addButton");
    const instance = wrapper.instance();

    instance.btnClick = jest.fn(instance.btnClick);

    button.simulate("click");
    expect(instance.btnClick).toMatchSnapshot();
  });
});