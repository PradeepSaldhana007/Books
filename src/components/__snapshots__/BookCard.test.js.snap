// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Executes a handler function must call the mock method with button click 1`] = `[MockFunction]`;
